ShardingSphere Agent is incompatible with Logback 1.5.22

## Bug Report

**For English only**, other languages will not accept.

Before report a bug, make sure you have:

- Searched open and closed [GitHub issues](https://github.com/apache/shardingsphere/issues).
- Read documentation: [ShardingSphere Doc](https://shardingsphere.apache.org/document/current/en/overview).

Please pay attention on issues you submitted, because we maybe need more details. 
If no response anymore and we cannot reproduce it on current information, we will **close it**.

Please answer these questions before submitting your issue. Thanks!

### Which version of ShardingSphere did you use?

- https://github.com/apache/shardingsphere/commit/18a1658bf4e8997e7e466de7fb08480c2bec9bc7

### Which project did you use? ShardingSphere-JDBC or ShardingSphere-Proxy?

- `ShardingSphere-JDBC` and `ShardingSphere-Agent`

### Expected behavior

- ShardingSphere Agent is compatible with Logback 1.5.22.

### Actual behavior

- ShardingSphere Agent is incompatible with Logback 1.5.22.

### Reason analyze (If you can)

- No sure.

### Steps to reproduce the behavior, such as: SQL to execute, sharding rule configuration, when exception occur etc.

- Execute the following commands on Ubuntu 24.04.4 WSL with `Docker Engine` and `SDKMAN!` installed.
```shell
sdk install java 23.0.2-graalce
sdk use java 23.0.2-graalce

git clone -b master git@github.com:apache/shardingsphere.git
cd ./shardingsphere/
git reset --hard 18a1658bf4e8997e7e466de7fb08480c2bec9bc7
./mvnw clean install -Prelease -T1C -DskipTests -Djacoco.skip=true -Dcheckstyle.skip=true -Drat.skip=true -Dmaven.javadoc.skip=true
./mvnw -am -pl distribution/agent -Prelease,docker -T1C -DskipTests clean package
cd ../

git clone git@github.com:linghengqian/shardingsphere-agent-master-test.git
cd ./shardingsphere-agent-master-test/
./mvnw clean package -T1C -DskipTests
docker compose up -d
docker compose logs --follow shardingsphere-agent-master-test
```

- The error log is as follows.
```shell
shardingsphere-agent-master-test-1  | Jan 06, 2026 3:17:04 AM org.apache.shardingsphere.agent.core.plugin.jar.PluginJarLoader load
shardingsphere-agent-master-test-1  | INFO: Loaded jar: shardingsphere-agent-logging-file-5.5.3-SNAPSHOT.jar
shardingsphere-agent-master-test-1  | Jan 06, 2026 3:17:04 AM org.apache.shardingsphere.agent.core.plugin.jar.PluginJarLoader load
shardingsphere-agent-master-test-1  | INFO: Loaded jar: shardingsphere-agent-plugin-core-5.5.3-SNAPSHOT.jar
shardingsphere-agent-master-test-1  | Jan 06, 2026 3:17:04 AM org.apache.shardingsphere.agent.core.plugin.jar.PluginJarLoader load
shardingsphere-agent-master-test-1  | INFO: Loaded jar: shardingsphere-agent-metrics-core-5.5.3-SNAPSHOT.jar
shardingsphere-agent-master-test-1  | Jan 06, 2026 3:17:04 AM org.apache.shardingsphere.agent.core.plugin.jar.PluginJarLoader load
shardingsphere-agent-master-test-1  | INFO: Loaded jar: shardingsphere-agent-metrics-prometheus-5.5.3-SNAPSHOT.jar
shardingsphere-agent-master-test-1  | Jan 06, 2026 3:17:04 AM org.apache.shardingsphere.agent.core.plugin.jar.PluginJarLoader load
shardingsphere-agent-master-test-1  | INFO: Loaded jar: shardingsphere-agent-tracing-opentelemetry-5.5.3-SNAPSHOT.jar
shardingsphere-agent-master-test-1  | SLF4J: Failed to load class "org.slf4j.impl.StaticLoggerBinder".
shardingsphere-agent-master-test-1  | SLF4J: Defaulting to no-operation (NOP) logger implementation
shardingsphere-agent-master-test-1  | SLF4J: See http://www.slf4j.org/codes.html#StaticLoggerBinder for further details.
shardingsphere-agent-master-test-1  | Exception in thread "main" java.lang.reflect.InvocationTargetException
shardingsphere-agent-master-test-1  |   at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:118)
shardingsphere-agent-master-test-1  |   at java.base/java.lang.reflect.Method.invoke(Method.java:580)
shardingsphere-agent-master-test-1  |   at org.springframework.boot.loader.launch.Launcher.launch(Launcher.java:106)
shardingsphere-agent-master-test-1  |   at org.springframework.boot.loader.launch.Launcher.launch(Launcher.java:64)
shardingsphere-agent-master-test-1  |   at org.springframework.boot.loader.launch.JarLauncher.main(JarLauncher.java:40)
shardingsphere-agent-master-test-1  | Caused by: java.lang.IllegalStateException: LoggerFactory is not a Logback LoggerContext but Logback is on the classpath. Either remove Logback or the competing implementation (class org.slf4j.helpers.NOPLoggerFactory loaded from file:/usr/agent/shardingsphere-agent.jar). If you are using WebLogic you will need to add 'org.slf4j' to prefer-application-packages in WEB-INF/weblogic.xml
shardingsphere-agent-master-test-1  |   at org.springframework.util.Assert.state(Assert.java:102)
shardingsphere-agent-master-test-1  |   at org.springframework.boot.logging.logback.LogbackLoggingSystem.getLoggerContext(LogbackLoggingSystem.java:420)
shardingsphere-agent-master-test-1  |   at org.springframework.boot.logging.logback.LogbackLoggingSystem.beforeInitialize(LogbackLoggingSystem.java:130)
shardingsphere-agent-master-test-1  |   at org.springframework.boot.context.logging.LoggingApplicationListener.onApplicationStartingEvent(LoggingApplicationListener.java:240)
shardingsphere-agent-master-test-1  |   at org.springframework.boot.context.logging.LoggingApplicationListener.onApplicationEvent(LoggingApplicationListener.java:222)
shardingsphere-agent-master-test-1  |   at org.springframework.context.event.SimpleApplicationEventMulticaster.doInvokeListener(SimpleApplicationEventMulticaster.java:180)
shardingsphere-agent-master-test-1  |   at org.springframework.context.event.SimpleApplicationEventMulticaster.invokeListener(SimpleApplicationEventMulticaster.java:173)
shardingsphere-agent-master-test-1  |   at org.springframework.context.event.SimpleApplicationEventMulticaster.multicastEvent(SimpleApplicationEventMulticaster.java:151)
shardingsphere-agent-master-test-1  |   at org.springframework.context.event.SimpleApplicationEventMulticaster.multicastEvent(SimpleApplicationEventMulticaster.java:133)
shardingsphere-agent-master-test-1  |   at org.springframework.boot.context.event.EventPublishingRunListener.multicastInitialEvent(EventPublishingRunListener.java:137)
shardingsphere-agent-master-test-1  |   at org.springframework.boot.context.event.EventPublishingRunListener.starting(EventPublishingRunListener.java:76)
shardingsphere-agent-master-test-1  |   at org.springframework.boot.SpringApplicationRunListeners.lambda$starting$0(SpringApplicationRunListeners.java:56)
shardingsphere-agent-master-test-1  |   at java.base/java.lang.Iterable.forEach(Iterable.java:75)
shardingsphere-agent-master-test-1  |   at org.springframework.boot.SpringApplicationRunListeners.doWithListeners(SpringApplicationRunListeners.java:123)
shardingsphere-agent-master-test-1  |   at org.springframework.boot.SpringApplicationRunListeners.starting(SpringApplicationRunListeners.java:56)
shardingsphere-agent-master-test-1  |   at org.springframework.boot.SpringApplication.run(SpringApplication.java:313)
shardingsphere-agent-master-test-1  |   at org.springframework.boot.SpringApplication.run(SpringApplication.java:1365)
shardingsphere-agent-master-test-1  |   at org.springframework.boot.SpringApplication.run(SpringApplication.java:1354)
shardingsphere-agent-master-test-1  |   at io.github.linghengqian.shardingsphere.agent.master.test.ShardingsphereAgentMasterTestApplication.main(ShardingsphereAgentMasterTestApplication.java:28)
shardingsphere-agent-master-test-1  |   at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:103)
shardingsphere-agent-master-test-1  |   ... 4 more
```

### Example codes for reproduce this issue (such as a github link).

- The full git repository is located at https://github.com/linghengqian/shardingsphere-agent-master-test .
- Early investigations involved https://github.com/apache/shardingsphere/issues/32793, https://github.com/apache/shardingsphere/issues/37412#issuecomment-3711276967, and https://github.com/apache/shardingsphere-elasticjob/issues/2425 .

--- a/agent/core/pom.xml
+++ b/agent/core/pom.xml
@@ -46,6 +46,11 @@
             <groupId>org.yaml</groupId>
             <artifactId>snakeyaml</artifactId>
         </dependency>
+        <dependency>
+            <groupId>org.slf4j</groupId>
+            <artifactId>slf4j-api</artifactId>
+            <scope>provided</scope>
+        </dependency>
     </dependencies>
     
     <build>

